searchcore/src/vespa/searchcore/proton/feedoperation/removeoperation.cpp

```diff
@@ -40,18 +39,69 @@ RemoveOperation::serialize(vespalib::nbostream &os) const
 
 
 void
-RemoveOperation::deserialize(vespalib::nbostream &is,
+RemoveOperationWithDocId::deserialize(vespalib::nbostream &is,
                           const DocumentTypeRepo &repo)
 {
-    DocumentOperation::deserialize(is, repo);
+    RemoveOperation::deserialize(is, repo);
     size_t oldSize = is.size();
     _docId = DocumentId(is);
     _serializedDocSize = oldSize - is.size();
 }
 
-vespalib::string RemoveOperation::toString() const {
+vespalib::string
+RemoveOperationWithDocId::toString() const {
     return make_string("Remove(%s, %s)",
-                       _docId.getScheme().toString().c_str(),
-                       docArgsToString().c_str());
+                       _docId.getScheme().toString().c_str(), docArgsToString().c_str());
+}
+
+RemoveOperationWithGid::RemoveOperationWithGid()
+        : RemoveOperation(FeedOperation::REMOVE_GID),
+          _gid(),
+          _docType(),
+          _lid(0)
+{}
+
+
+RemoveOperationWithGid::RemoveOperationWithGid(BucketId bucketId, Timestamp timestamp, const GlobalId &gid, vespalib::stringref docType, uint32_t lid)
+        : RemoveOperation(FeedOperation::REMOVE_GID, bucketId, timestamp),
+          _gid(gid),
+          _docType(docType),
+          _lid(lid)
+{}
+
+RemoveOperationWithGid::~RemoveOperationWithGid() = default;
+
+void
+RemoveOperationWithGid::serialize(vespalib::nbostream &os) const
+{
+    assertValidBucketId(_gid);
+    RemoveOperation::serialize(os);
```

Comment:
Pleas add an `assert(!getValidDbdId());`, to ensure that we're not serializing a bad RemoveOperationWithGid.

Reply:
Fixed